nym-vpn-x/src-tauri/src/tray.rs

```diff
@@ -0,0 +1,149 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use strum::AsRefStr;
+use tauri::image::Image;
+use tauri::menu::MenuEvent;
+use tauri::tray::TrayIconEvent;
+use tauri::tray::{MouseButton, MouseButtonState};
+use tauri::tray::{TrayIcon, TrayIconBuilder};
+use tauri::{include_image, Manager};
+use tauri::{menu::MenuBuilder, AppHandle};
+use tracing::{debug, error, instrument, trace, warn};
+
+#[cfg(not(target_os = "linux"))]
+use crate::APP_NAME;
+use crate::{
+    grpc::client::GrpcClient,
+    states::{app::ConnectionState, SharedAppState},
+    window::AppWindow,
+    MAIN_WINDOW_LABEL,
+};
+
+pub const TRAY_ICON_ID: &str = "main";
+pub const TRAY_MENU_ID: &str = "tray_menu";
+const APP_ICON: Image<'_> = include_image!("icons/icon.png");
+
+#[derive(AsRefStr, Debug)]
+enum MenuItemId {
+    ShowHide,
+    Quit,
+}
+
+#[instrument(skip_all)]
+fn on_tray_event(tray_icon: &TrayIcon, event: TrayIconEvent) {
+    if let TrayIconEvent::Click {
+        button: MouseButton::Left,
+        button_state: MouseButtonState::Down,
+        ..
+    } = event
+    {
+        trace!("tray event left click");
+        show_window(tray_icon.app_handle(), false).ok();
+    }
+}
+
+#[instrument(skip(app))]
+fn on_menu_event(app: &AppHandle, event: MenuEvent) {
+    trace!("menu event [{}]", event.id.0);
+
+    match event.id().as_ref() {
+        "ShowHide" => {
```

Comment:
Make these constants.

Reply:
Why we'd need that, what value it brings to? Those variants are only relevant in this context of pattern matching (aka the `match` right?).

Maybe you missed the enum above?
```
#[derive(AsRefStr, Debug)]
enum MenuItemId {
    ShowHide,
    Quit,
}
```

Reply:
If it's effectively matching over the menu item enum, then a possible way here would be to construct the enum from the event

Reply:
@doums because compiler will not warn you if tomorrow you or someone else remove or rename any of `MenuItemId` variants. If you were to match against `MenuItemId` then the compiler would error.